src/Analyzer/FileVisitor.php

```diff
@@ -28,12 +28,15 @@ public function enterNode(Node $node): void
             }
         }
 
-        if ($node instanceof Node\Expr\Instanceof_) {
+        if ($node instanceof Node\Expr\Instanceof_ && method_exists($node->class, 'toString')) {
             $this->classDescriptionBuilder
                 ->addDependency(new ClassDependency($node->class->toString(), $node->getLine()));
         }
 
-        if ($node instanceof Node\Expr\New_ && !($node->class instanceof Node\Expr\Variable)) {
+        if ($node instanceof Node\Expr\New_ &&
+            !($node->class instanceof Node\Expr\Variable) &&
+            method_exists($node->class, 'toString')
```

Comment:
Should we extract a method for this particular condition? It find it a bit hard to get

Reply:
We can't move all conditions in another function for psalm reasons.
But I have improved a little bit.